File: August2022/Java/PowerOfFour.java

// @saorav21994
// TC : O(log n base 4)
// SC : O(1)

class Solution {
    public boolean isPowerOfFour(int n) {
        double loggy = Math.log(n)/Math.log(4);
        int ilog = (int)loggy;
        loggy = Math.ceil(loggy);
        return (loggy==ilog);
    }
}

File: August2022/C++/PowerOfFour.cpp
class Solution {
public:
    // TC: O(logn)
    // SC: O(logn)
    bool isPowerOfFour(int n) {
        if (n == 1) {
            return true;
        }
        if (n % 4 == 0 && n != 0) {
            return isPowerOfFour(n / 4);
        } else {
            return false;
        }
    }
};
